tests/pytest_tests/unit_tests/test_artifacts/test_storage.py

```diff
@@ -345,3 +346,17 @@ class UnfinishedStorageHandler(wandb_artifacts.StorageHandler):
         ush.load_path(manifest_entry=None)
     with pytest.raises(NotImplementedError):
         ush.store_path(artifact=None, path="")
+
+
+def test_unwritable_staging_dir(monkeypatch):
+    # Use a non-writable directory as the staging directory.
+    # CI just doesn't care about permissions, so we're patching os.makedirs 🙃
+    def nope(*args, **kwargs):
+        raise OSError(13, "Permission denied")
+
+    monkeypatch.setattr(os, "makedirs", nope)
+
+    with pytest.raises(PermissionError) as excinfo:
```

Comment:
you can use match and then you don't need to write the assertion (obviously stylistic choice)

Reply:
I actually forgot that `pytest.raises` accepted a match parameter!